rd_ui/app/scripts/controllers/query_view.js

```diff
@@ -230,12 +230,20 @@
       $scope.$parent.pageTitle = $scope.query.name;
     });
 
+    var firstLoad = true;
     $scope.$watch('queryResult && queryResult.getData()', function(data, oldData) {
       if (!data) {
         return;
       }
 
       $scope.filters = $scope.queryResult.getFilters();
+      if (firstLoad) {
+        _.each($scope.filters, function (queryFilter) {
+          if (_.has($location.search(), queryFilter.name))
+            queryFilter.current = $location.search()[queryFilter.name];
```

Comment:
As far as I can tell, only searching for queries updates the URI params.

However, I agree that updating it is better than not.  I'll look into it.